flink-core/src/main/java/org/apache/flink/configuration/StateBackendOptions.java

```diff
@@ -0,0 +1,95 @@
+    //  general state backend options
+    // ------------------------------------------------------------------------
+
+    /**
+     * The checkpoint storage used to store operator state locally within the cluster during
+     * execution.
+     *
+     * <p>The implementation can be specified either via their shortcut name, or via the class name
+     * of a {@code StateBackendFactory}. If a StateBackendFactory class name is specified, the
+     * factory is instantiated (via its zero-argument constructor) and its {@code
+     * StateBackendFactory#createFromConfig(ReadableConfig, ClassLoader)} method is called.
+     *
+     * <p>Recognized shortcut names are 'hashmap' and 'rocksdb'.
+     */
+    @Documentation.Section(value = Documentation.Sections.COMMON_STATE_BACKENDS, position = 1)
+    public static final ConfigOption<String> STATE_BACKEND =
+            ConfigOptions.key("state.backend")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            Description.builder()
+                                    .text("The state backend to be used to store state.")
+                                    .linebreak()
+                                    .text(
+                                            "The implementation can be specified either via their shortcut "
+                                                    + " name, or via the class name of a %s. "
+                                                    + "If a factory is specified it is instantiated via its "
+                                                    + "zero argument constructor and its %s "
+                                                    + "method is called.",
+                                            TextElement.code("StateBackendFactory"),
+                                            TextElement.code(
+                                                    "StateBackendFactory#createFromConfig(ReadableConfig, ClassLoader)"))
+                                    .linebreak()
+                                    .text("Recognized shortcut names are 'hashmap' and 'rocksdb'.")
+                                    .build());
+
+    @Documentation.Section(Documentation.Sections.COMMON_STATE_BACKENDS)
+    public static final ConfigOption<Boolean> LATENCY_TRACK_ENABLED =
+            ConfigOptions.key("state.backend.latency-track-enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Whether to track latency of state operations, e.g value state put/get/clear.");
+
+    @Documentation.Section(Documentation.Sections.COMMON_STATE_BACKENDS)
+    public static final ConfigOption<Integer> LATENCY_TRACK_SAMPLE_INTERVAL =
+            ConfigOptions.key("state.backend.latency-track-sample-interval")
+                    .intType()
+                    .defaultValue(100)
```

Comment:
I'd either use `durationType` (or similar) or specify unit in the key name. I always end up looking up the unit in such cases :)

Reply:
This is not a time based configuration, I don't think `durationType` could describe it.

Reply:
I see.